formapp/app.py

```diff
@@ -690,25 +694,18 @@ def updateStep():
 
 '''
     /
-    Homepage (root)
+    Redirect to either basic or advanced mode based on the user's preference
 '''
 @app.route('/')
-def endpoint_home():
-        if not request.remote_addr in sD.studentDict:
-            return redirect('/login')
-        else:
-            if True:
-                username = sD.studentDict[request.remote_addr]['name']
-                sfx.updateFiles()
-                sounds = []
-                music = []
-                for key, value in sfx.sound.items():
-                    sounds.append(key)
-                for key, value in bgm.bgm.items():
-                    music.append(key)
-                return render_template('index.html', username = username, sfx = sounds, bgm = music)
-            else:
-                redirect('/basic')
+def endpoint_root():
+    ##Also check database
+    if not request.remote_addr in sD.studentDict:
+        return redirect('/login')
+    if sD.studentDict[request.remote_addr]['preferredMode'] == 'advanced':
+        return redirect('/home')
+    if sD.studentDict[request.remote_addr]['preferredMode'] == 'basic':
+        return redirect('/basic')
+    return redirect('/setdefault')
```

Comment:
Instead of redirected to a new endpoint, you can just render the correct template.

Reply:
I'm using redirects because:
1. For advanced mode, it needs to get the username, sfx, and bgm before rendering the template.
2. I want the URL to match what page the user sees